lib/services/lookup.js

```diff
@@ -482,6 +482,31 @@ function lookupServiceFactory(
     };
 
     /**
+     * Returns a list of object containing IP and Mac Addresses for the given Node ID.
+     * @param  {String} id Node ID
+     * @return {Promise.<Array>} A promise fulfilled with an array of
+     * objects containing IP Addresses and Mac Addresses associated with the Node.
+     */
+    LookupService.prototype.findIpMacAddresses = function (id) {
+        var self = this;
+        assert.isMongoId(id);
+
+        return self.validateArpCache()
+        .then(function() {
+            return waterline.lookups.findByTerm(id).then(function (records) {
```

Comment:
I don't think it's a good idea to use `findByTerm` as the `id` here is absolutely node Id. So use `waterline.lookups.find({ node: id})` will not only gain performance but also avoid a potentially bug.
If you look at the implementation of `findByTerm`, there is some 'smart` logic inside it, the input `id` may be an IP address, a MAC address or a nodeId, this function smartly detects the format of your input. But in your case, this kind of 'smart detection' is not necessary and may cause the bug that it thinks the id is MAC address rather than node id (especially you set a random node id in unit-test).